modules/hdf4_handler/HDFStructure.cc

```diff
@@ -79,6 +81,59 @@ void HDFStructure::set_read_p(bool state) {
 	BaseType::set_read_p(state);
 }
 
+bool HDFStructure::read_from_value(vector<uint8_t> &values, size_t &values_offset) {
+
+#if 0
+    Constructor::Vars_iter vi = this->var_begin();
+    Constructor::Vars_iter ve = this->var_end();
+
+    for (; vi != ve; vi++) {
+#endif
+
+    for (auto &bt:this->variables()) {	     
+        //BaseType *bt = *vi;
```

Comment:
Sonar scan will probably complain about this commented out code. Also, it would be good to remove the #if 0...#endif blocks if we're unlikely to need them again.

Reply:
Yes, correct it.